pytorch_lightning/loops/epoch/evaluation_epoch_loop.py

```diff
@@ -40,9 +40,17 @@ def __init__(self) -> None:
         self._dl_max_batches: Optional[int] = None
         self._num_dataloaders: Optional[int] = None
         self.outputs: EPOCH_OUTPUT = []
-        self.batch_progress = Progress()
+        self.batch_progress = BatchProgress()
         self.dataloader_iter: Optional[Iterator] = None
 
+    @property
+    def has_completed_validation_batch(self) -> bool:
+        return self.batch_progress.current.ready == self.batch_progress.current.completed
+
+    @property
+    def has_completed_dataloader(self) -> bool:
+        return self.has_completed_validation_batch and self.batch_progress.is_last_batch
```

Comment:
For consistency with the `TrainingEpochLoop`, maybe we should have

```suggestion
    @property
    def _num_val_batches_reached(self) -> bool:
        return self.batch_progress.current.ready == self.trainer.num_val_batches self.batch_progress.is_last_batch
```

Reply:
Done !